### ISSUE TITLE
Add Additional Link Shape Properties

### ISSUE THREAD

**Issue by sanbales** [jupyrdf/ipyforcegraph#60]
## Elevator Pitch

Add `line_dash` and `curvature` to the `links`.

## Motivation

* `curvature` and `line dashes` could be used to convey information about relationships.
* would produce a more complete 1:1 with the `force-graph` API.

## Design Ideas
* For curvature it would be pretty straight forward from the `python` side.
  ```python
  curvature: TNumFeature = _make_trait("the curvature of the link, 0: straight, 1: circular", numeric=True)
  ```
* The `line_dash` presents a more complicated challenge, as the data is really a list of numbers, as per the [`HTML Canvas setLineDash`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/setLineDash) method.
  ```python
  line_dash: TListNumFeature = _make_trait(
      "the dash pattern, e.g., [5, 15] to draw a repeating pattern of a 5 pixel segment followed by a 15 pixel blank",
      numeric=True,
      is_list=True,
  )
  ```

**Comment by sanbales** [comment]
2197523bf5659958a6a1ab767ccbbc0147eaaa70 adds `curvature` without any issues
![image](https://user-images.githubusercontent.com/1438114/224494633-85c0f5e2-e88c-4df5-9dba-1ff53981f84d.png)

**Comment by sanbales** [comment]
#62 seems to be working well

* `link_shape.curvature = 0.2`
* `link_shape.line_dash.value = "[5, 3]"`
* `link_shape.weight = 2`

![image](https://user-images.githubusercontent.com/1438114/224508484-4e24e0e1-355c-4d6f-b92e-411ef43f543c.png)

**Comment by sanbales** [comment]
Resolved in #62, future work could add `rotation` but that only works on `ForceGraph3D`.
